lib/xml_utils.pm

```diff
@@ -51,26 +51,36 @@ Verifies that node by given XPath is unique and has expected value. C<%args> is
 
 =item * C<xpath> - XPath to the node which value we want to check
 
-=item * C<expected_val> - expected value for the node
+=item * C<expected_value> - expected value for the node`
+
+=item * C<search_by_value> - search is performed by value instead of by node, which is default`
 
 =back
 
 =cut
 
 sub verify_option {
     my (%args) = @_;
-
-    my @nodes = find_nodes(%args);
-    ## Verify that there is node found by xpath and it's single one
-    if (scalar @nodes != 1) {
-        return "Generated autoinst.xml contains unexpected number of nodes for xpath: $args{xpath}. Found: " . scalar @nodes . ", expected: 1.";
+    if ($args{search_by_value}) {
+        my $value = $args{xpc}->findvalue($args{xpath});
+        if ($value ne $args{expected_value}) {
+            return "Unexpected value for xpath $args{xpath}. Expected: '$args{expected_value}', got: '$value'";
```

Comment:
"Generated autoinst.xml" would be useful here to give some extra information where the error come from.

Reply:
I kept this message and the logic below.

Reply:
I see. resolve this conversation